### PR TITLE
feat: support `parseForESLint` from custom parser

### LINKED ISSUE #1433: Parse error in mdx files
Hi, I'm reporting here too, because I realized [an issue](https://github.com/tleunen/eslint-import-resolver-babel-module/issues/106) I originally reported at `babel-plugin-module-resolver` might not be specific to that resolver. Please read it for more information. 

As I commented there, I also get parse errors when using the default node resolver, but on a different import.

import:
```jsx 
import styled from "styled-components";
```

Error:
```
21:20  error    Parse errors in imported module 'styled-components': parser.parse is not a function (undefined:undefined)  import/namespace
21:20  error    Parse errors in imported module 'styled-components': parser.parse is not a function (undefined:undefined)  import/default
21:20  warning  Parse errors in imported module 'styled-components': parser.parse is not a function (undefined:undefined)  import/no-named-as-default
21:20  warning  Parse errors in imported module 'styled-components': parser.parse is not a function (undefined:undefined)  import/no-named-as-default-member
```

### PR DESCRIPTION
**Pull request by JounQin**
close #1433

### CODE REVIEW

**Comment by cdfa** [comment]
So I ran `npm i -D JounQin/eslint-plugin-import#feat/parse_for_eslint`, but it's not working. I looked for the `utlis/parse.js` file you modified and it seems to be located at `node_modules/eslint-module-utils/parse.js`, but it doesn't have your patch. How do I test it?

**Comment by JounQin** [comment]
@cdfa There is no need to wait for this PR, I've exported `parse` from `@rxts/eslint-plugin-mdx`, you can simply upgrade to the latest version.

**Comment by cdfa** [comment]
Ah great!

File: tests/src/core/parse.js [hunk 1]
```diff
@@ -43,6 +45,16 @@ describe('parse(content, { settings, ecmaFeatures })', function () {
     expect(parseSpy.args[0][1], 'custom parser to get parserOptions.filePath equal to the full path of the source file').to.have.property('filePath', path)
   })
 
+  it('passes with custom `parseForESLint` parser', function () {
+    const parseForESLintSpy = sinon.spy(eslintParser.parseForESLint)
```
Comment (ljharb) [inline tests/src/core/parse.js]: please avoid passing just an object into `sinon.spy`; that implicitly spies on all properties. Instead, let's pass explicit property names.

File: utils/parse.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-"use strict"
+'use strict'
```
Comment (ljharb) [inline utils/parse.js]: please revert all unrelated formatting changes

File: tests/src/core/parse.js (hunk 1 above)
Comment (JounQin) [reply tests/src/core/parse.js]: It's a function

File: utils/parse.js (hunk 2 above)
Comment (JounQin) [reply utils/parse.js]: OK, it must be related to my editor `formatOnSave`.

Besides, why the usage of quotes are inconsistent through the project.

File: tests/src/core/parse.js (hunk 1 above)
Comment (ljharb) [reply tests/src/core/parse.js]: i'm a bit confused; sinon is for spying on properties. what does passing the function in here do, as opposed to `sinon.spy(eslintParser, 'parseForEslint')`?

File: utils/parse.js (hunk 2 above)
Comment (ljharb) [reply utils/parse.js]: #1413; however, either way, what eslint doesn't make consistent, shouldn't be autoformatted :-)

File: tests/src/core/parse.js (hunk 1 above)
Comment (JounQin) [reply tests/src/core/parse.js]: It can spy function (wrapping the original one) directly. The tests are passing.😆

Comment (ljharb) [reply tests/src/core/parse.js]: Iiiiiii

Comment (ljharb) [reply tests/src/core/parse.js]: right, but if you followed my suggestion, you wouldn’t have to reassign it on the next lines.

Comment (JounQin) [reply tests/src/core/parse.js]: > right, but if you followed my suggestion, you wouldn’t have to reassign it on the next lines.

Seems more clear, I will change it.

File: tests/src/core/parse.js [hunk 3]
```diff
@@ -43,6 +45,15 @@ describe('parse(content, { settings, ecmaFeatures })', function () {
     expect(parseSpy.args[0][1], 'custom parser to get parserOptions.filePath equal to the full path of the source file').to.have.property('filePath', path)
   })
 
+  it('passes with custom `parseForESLint` parser', function () {
+    const parseForESLintSpy = sinon.spy(eslintParser, 'parseForESLint')
+    const parseSpy = sinon.spy()
+    eslintParser.parse = parseSpy
```
Comment (ljharb) [inline tests/src/core/parse.js]: this can also be `const parseSpy = sinon.spy(eslintParser, 'parse');`

Comment (JounQin) [reply tests/src/core/parse.js]: > this can also be `const parseSpy = sinon.spy(eslintParser, 'parse');`

Nope, `eslintParser.parse` is `undefined`, `sinon` will throw an error.

See https://travis-ci.org/benmosher/eslint-plugin-import/builds/567436253?utm_source=github_status&utm_medium=notification

Comment (ljharb) [reply tests/src/core/parse.js]: ah, fair enough :-)
